rules/channel_constraints.go

```diff
@@ -0,0 +1,354 @@
+package rules
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/lightninglabs/lightning-terminal/firewalldb"
+	"github.com/lightninglabs/lightning-terminal/litrpc"
+	mid "github.com/lightninglabs/lightning-terminal/rpcmiddleware"
+	"github.com/lightninglabs/lightning-terminal/session"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	// Compile-time checks to ensure that ChannelConstraint and
+	// ChanConstraintMgr implement the appropriate Manager, Enforcer and
+	// Values interface.
+	_ Manager  = (*ChanConstraintMgr)(nil)
+	_ Enforcer = (*ChannelConstraint)(nil)
+	_ Values   = (*ChannelConstraint)(nil)
+)
+
+// ChanConstraintName is the string identifier of the ChannelConstraint rule.
+const ChanConstraintName = "channel-constraint"
+
+// ChanConstraintMgr manages the ChannelConstraint rule.
+type ChanConstraintMgr struct{}
+
+// Stop cleans up the resources held by the manager.
+//
+// NOTE: This is part of the Manager interface.
+func (m *ChanConstraintMgr) Stop() error {
+	return nil
+}
+
+// NewEnforcer constructs a new ChannelConstraint rule enforcer using the passed
+// values and config.
+//
+// NOTE: This is part of the Manager interface.
+func (m *ChanConstraintMgr) NewEnforcer(_ Config, values Values) (Enforcer,
+	error) {
+
+	bounds, ok := values.(*ChannelConstraint)
+	if !ok {
+		return nil, fmt.Errorf("values must be of type "+
+			"ChannelConstraint, got %T", values)
+	}
+
+	return bounds, nil
+}
+
+// NewValueFromProto converts the given proto value into a ChannelConstraint
+// Value object.
+//
+// NOTE: This is part of the Manager interface.
+func (m *ChanConstraintMgr) NewValueFromProto(value *litrpc.RuleValue) (Values,
+	error) {
+
+	rv, ok := value.Value.(*litrpc.RuleValue_ChannelConstraint)
+	if !ok {
+		return nil, fmt.Errorf("incorrect RuleValue type %T",
+			value.Value)
+	}
+
+	channelBounds := rv.ChannelConstraint
+
+	return &ChannelConstraint{
+		MinCapacitySat: channelBounds.MinCapacitySat,
+		MaxCapacitySat: channelBounds.MaxCapacitySat,
+		MaxPushSat:     channelBounds.MaxPushSat,
```

Comment:
missing the new public/private allowed?

Reply:
i think both are false here meaning no channels allowed?

should we make the default (false) => allowed & then rather have "NoPublic"/"NoPrivate"?

Reply:
does a call to OpenChan work with this as is given that both are false here?

Reply:
> does a call to OpenChan work with this as is given that both are false here?

I think `UnmarshalRuleValues` is only used for sanity checking upon session registration, but the actual used final rule values are the ones from the server, so even if those are false, channel openings work.